src/shogun/mathematics/linalg/LinalgNamespaceMatrix.cpp

```diff
@@ -0,0 +1,43 @@
+#include <shogun/lib/DataType.h>
+#include <shogun/mathematics/linalg/LinalgNamespaceMatrix.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+#define DEFINE_SWITCH(PTYPE, FUNC, ...) \
+switch (PTYPE) \
+{ \
+	case PT_FLOAT32: \
+		FUNC<float32_t>(__VA_ARGS__); \
```

Comment:
cool! That is indeed the easiest solution for what we have in mind
There is this issue:
 * Imagine you have some rather cheap operations (small dim add)
 * imagine you call this all the time, i.e. in a loop
 * Then the switch might have some run-time drawbacks

There is two competing mindsets here:
 * We can say "we are not in rush to compute" (@lisitsyn) and argue that looped linalg calls should rather be put into a vectorized linalg method that doesnt do the switch
 * We can add a mechanism that kind of "tells" linalg that the next calls are all say `float64_t` (and the switch is ignored)
 * the above could work via registering pointers to the implementations

These have to be traded off. I rather tend to the first solution